forta-bots/No-Settle/src/agent.ts

```diff
@@ -0,0 +1,77 @@
+import { Finding, getEthersProvider, HandleBlock, BlockEvent, Initialize } from "forta-agent";
+import { providers, BigNumber } from "ethers";
+import NetworkManager, { NETWORK_MAP } from "./network";
+import NetworkData from "./network";
+import { createFinding, DAYS_THRESHOLD, SETTLE_IFACE } from "./utils";
+
+const networkManager: NetworkManager = new NetworkManager(NETWORK_MAP);
+let latestSettleTimestamp: BigNumber = BigNumber.from(0);
+
+export const provideInitialize =
+  (data: NetworkManager, provider: providers.Provider, daysThreshold: number): Initialize =>
+  async () => {
+    const { chainId } = await provider.getNetwork();
+    data.setNetwork(chainId);
+    const blockNumber: number = await provider.getBlockNumber();
+
+    const filter = {
+      address: data.TeleportJoin,
+      topics: [SETTLE_IFACE.getEventTopic("Settle")],
+      fromBlock: blockNumber - daysThreshold * 6100, // 6100 blocks/day on average
+      toBlock: blockNumber,
+    };
+
+    const settleEvents: providers.Log[] = await provider.getLogs(filter);
+
+    if (settleEvents.length) {
+      latestSettleTimestamp = BigNumber.from(
+        (await provider.getBlock(settleEvents[settleEvents.length - 1].blockNumber)).timestamp
+      );
+    }
+  };
+
+export const provideHandleBlock = (
+  data: NetworkData,
+  provider: providers.Provider,
+  daysThreshold: number
+): HandleBlock => {
+  let latestAlertedSettleTimestamp: BigNumber = BigNumber.from(-1);
+  return async (blockEvent: BlockEvent) => {
+    const findings: Finding[] = [];
+    const currentTimestamp: BigNumber = BigNumber.from(blockEvent.block.timestamp);
+
+    // Check if:
+    // 1) Alert has not already been created for this settle timestamp
+    // 2) Threshold has been exceeded
+    if (
+      !latestAlertedSettleTimestamp.eq(latestSettleTimestamp) &&
+      currentTimestamp.sub(latestSettleTimestamp).gt(BigNumber.from(daysThreshold).mul(86400))
```

Comment:
tiny suggestion:
you can store `BigNumber.from(daysThreshold).mul(86400)` in a const var, above line 39.